src/common/transformations/include/transformations/rt_info/nms_selected_indices.hpp

```diff
@@ -21,7 +21,7 @@ TRANSFORMATIONS_API bool has_nms_selected_indices(const Node* node);
 
 TRANSFORMATIONS_API void set_nms_selected_indices(Node* node);
 
-class TRANSFORMATIONS_API NmsSelectedIndices : ov::RuntimeAttribute {
+class TRANSFORMATIONS_API NmsSelectedIndices : public ov::RuntimeAttribute {
```

Comment:
I know the answer now :D

Reply:
What about to share knowledge with other guys? 😄

Reply:
The code itself should be fine without change it looks that it can be some kind of bug MSVC in specific version when building with cpp20.

The compilation error on MSVC is related to `ov::Any` [base_type_info_impl](https://github.com/openvinotoolkit/openvino/blob/28ff6cef3b084c59f41b7b536938ffbb43868a27/src/core/include/openvino/core/any.hpp#L590), the helper `HasBaseMemberType` is incorrectly resolved on this compiler.

As `NmsSelectedIndices` hase `Base` as private it should be resolved to `false` and it is correctly done using gcc, clang and on my local machine using `MSVC 19.29.30146.0`, but on CI `MSVC 19.38.33130.0` is used which resolve this value as `true`. As result in this [line](https://github.com/openvinotoolkit/openvino/blob/28ff6cef3b084c59f41b7b536938ffbb43868a27/src/core/include/openvino/core/any.hpp#L592) is error because try access private member `Base`.

This change is like workaround for MSVC build, but it looks like not break anything so I think can be used.
But it changes the `NmsSelectedIndices` implementation and how `ov::Any` handle it.